eZ/Publish/API/Repository/Tests/SearchService/SearchServiceFulltextEmbedTest.php

```diff
@@ -0,0 +1,184 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Publish\API\Repository\Tests\SearchService;
+
+use eZ\Publish\API\Repository\SearchService;
+use eZ\Publish\API\Repository\Tests\BaseTest;
+use eZ\Publish\API\Repository\Values\Content\LocationQuery;
+use eZ\Publish\API\Repository\Values\Content\Query;
+use eZ\Publish\API\Repository\Values\Content\Query\Criterion;
+use eZ\Publish\API\Repository\Values\Content\Search\SearchHit;
+use eZ\Publish\Core\FieldType\RichText\Value as RichTextValue;
+use eZ\Publish\Core\Repository\Values\Content\Content;
+use eZ\Publish\Core\Repository\Values\Content\ContentCreateStruct;
+
+/**
+ * Test case for full text search in the SearchService (for embed).
+ *
+ * @see \eZ\Publish\API\Repository\SearchService
+ * @group integration
+ * @group search
+ * @group fulltext
+ */
+class SearchServiceFulltextEmbedTest extends BaseTest
+{
+    private const EMBEDDED_ARTICLE_NAME = 'test1';
+
+    private static $createdIds = [];
+
+    protected function setUp()
+    {
+        parent::setUp();
+
+        $repository = $this->getRepository(false);
+
+        if (false === $repository->getSearchService()->supports(SearchService::CAPABILITY_ADVANCED_FULLTEXT)) {
+            $this->markTestSkipped('Advanced FullText search is not supported by the current search engine');
+        }
+    }
+
+    public function testPrepareContent(): void
+    {
+        $contentService = $this->getRepository()->getContentService();
+        $baseArticleStruct = $this->prepareBaseArticleStruct();
+
+        $embeddedArticleStruct = $this->fillEmbeddedArticleStruct(clone $baseArticleStruct);
+        $embeddedArticleContent = $contentService->publishVersion(
+            $this->createContent($embeddedArticleStruct)->versionInfo
+        );
+
+        $mainArticleStruct = $this->fillMainArticleStruct(clone $baseArticleStruct, $embeddedArticleContent->id);
+        $mainArticleContent = $contentService->publishVersion(
+            $this->createContent($mainArticleStruct)->versionInfo
+        );
+
+        $this->refreshSearch($this->getRepository());
+
+        self::$createdIds = [
```

Comment:
This will makes a duplicates in search results as mentioned before, but if you want I can do that this way

Reply:
> This will makes a duplicates in search results as mentioned before, but if you want I can do that this way or add some additional code to avoid this

No, it should not, something is wrong then. Both database data and search index are independent per test method. You need to drop `@depends`. Otherwise the outcome might be random - depending on other tests executed in the suite.